pkg/security/keychain.go

```diff
@@ -58,14 +67,176 @@ func SecKeyUpdate(connectionID string, username string, password string) *SecErr
 }
 
 // SecKeyGetSecret : retrieve secret / credentials from the keyring
-func SecKeyGetSecret(connectionID string, username string) (string, *SecError) {
-
+func SecKeyGetSecret(connectionID, username string) (string, *SecError) {
 	conID := strings.TrimSpace(strings.ToLower(connectionID))
 	uName := strings.TrimSpace(strings.ToLower(username))
+	secret, err := GetSecretFromKeyring(conID, uName)
+	if err != nil {
+		return "", err
+	}
+	return secret, nil
+}
+
+// StoreSecretInKeyring stores the secret in either the system keyring or our insecure keyring.
+func StoreSecretInKeyring(connectionID, uName, pass string) *SecError {
+	service := connectionIDToService(connectionID)
+	if globals.UseInsecureKeyring {
+		_, statErr := os.Stat(GetPathToInsecureKeyring())
+		if os.IsNotExist(statErr) {
+			os.MkdirAll(insecureKeyringDir, 0777)
+			os.OpenFile(GetPathToInsecureKeyring(), os.O_CREATE, 0666)
```

Comment:
Do you want to give permissions to make this world readable? I think you might be able to stick to user readable only with 0600.